diff --git a/src/libasr/runtime/lfortran_intrinsics.c b/src/libasr/runtime/lfortran_intrinsics.c
index 954900615..cb9110c21 100644
--- a/src/libasr/runtime/lfortran_intrinsics.c
+++ b/src/libasr/runtime/lfortran_intrinsics.c
@@ -19,8 +19,8 @@
 #  include <unistd.h>
 #endif
 
-#include <libasr/runtime/lfortran_intrinsics.h>
-#include <libasr/config.h>
+#include "./lfortran_intrinsics.h"
+#include "./../config.h"
 
 #ifdef HAVE_RUNTIME_STACKTRACE
 
diff --git a/src/runtime/legacy/CMakeLists.txt b/src/runtime/legacy/CMakeLists.txt
index 3795d800c..f7e66d297 100644
--- a/src/runtime/legacy/CMakeLists.txt
+++ b/src/runtime/legacy/CMakeLists.txt
@@ -1,5 +1,5 @@
 set(SRC
-    ../../../src/libasr/runtime/lfortran_intrinsics.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/../../../src/libasr/runtime/lfortran_intrinsics.c
 )
 
 if(WIN32)
