diff --git a/src/runtime/legacy/CMakeLists.txt b/src/runtime/legacy/CMakeLists.txt
index 3795d800c..4f4f24c1c 100644
--- a/src/runtime/legacy/CMakeLists.txt
+++ b/src/runtime/legacy/CMakeLists.txt
@@ -1,5 +1,5 @@
 set(SRC
-    ../../../src/libasr/runtime/lfortran_intrinsics.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/../../../src/libasr/runtime/lfortran_intrinsics.c
 )

 if(WIN32)
@@ -32,7 +32,7 @@ if(WITH_TARGET_WASM)
     endif()

     add_custom_command(OUTPUT lfortran_runtime_wasm.o
-        COMMAND $ENV{WASI_SDK_PATH}/bin/clang -I${libasr_SOURCE_DIR}/.. -DCOMPILE_TO_WASM
+        COMMAND $ENV{WASI_SDK_PATH}/bin/clang -I${CMAKE_CURRENT_SOURCE_DIR}/../.. -DCOMPILE_TO_WASM
             -D_WASI_EMULATED_PROCESS_CLOCKS -c --target=wasm32-wasi ${SRC} -o lfortran_runtime_wasm.o
         COMMENT "Cross compiling lfortran_intrinscs.c to lfortran_runtime_wasm.o"
         DEPENDS ${SRC})
