diff --git a/src/libasr/CMakeLists.txt b/src/libasr/CMakeLists.txt
index aaf9db9..e5ce172 100644
--- a/src/libasr/CMakeLists.txt
+++ b/src/libasr/CMakeLists.txt
@@ -12,7 +12,8 @@ if (NOT LFORTRAN_VERSION)
         CACHE STRING "LFortran version" FORCE)
 endif ()
 
-configure_file(config.h.in config.h)
+message(STATUS "OOO CMAKE_CURRENT_DIR ${CMAKE_CURRENT_DIR}")
+configure_file(config.h.in ${CMAKE_CURRENT_SOURCE_DIR}/libasr/config.h)
 
 set(SRC
     codegen/asr_to_cpp.cpp
